src/halmos/processes.py

```diff
@@ -0,0 +1,217 @@
+import concurrent.futures
```

Comment:
new module that provides an executor/futures abstraction layer built on top of subprocess.Popen